zebra/zebra_nhg.c

```diff
@@ -99,31 +99,53 @@ nhg_connected_tree_root(struct nhg_connected_tree_head *head)
 	return nhg_connected_tree_first(head);
 }
 
-void nhg_connected_tree_del_nhe(struct nhg_connected_tree_head *head,
-				struct nhg_hash_entry *depend)
+struct nhg_hash_entry *
```

Comment:
I'm ... not following the change here - could you add a comment block that explains what is being returned when, and who's responsible for the memory?

Reply:
sure

Reply:
see `zebra_nhg_private.h` for info on return values